frontend/src/components/common/Tab.tsx

```diff
@@ -20,25 +20,40 @@ interface TabsProps {
   tabProps?: {
     [propName: string]: any;
   };
+  isDefaultSelectedRequired?: boolean;
```

Comment:
The title of this patch should be less generic. Also, if you need to pass it a default index or nothing, why not just use `number | null` and watch if that property changes so you can also control it from outside?